giotto/mapper/utils/visualization.py

```diff
@@ -27,50 +28,111 @@ def set_node_sizeref(node_elements, node_scale=12):
-
-            column_color_buttons.append(
-                dict(
-                    args=[{
-                        'marker.color': [None, node_color],
-                        'marker.cmin': [None, np.min(node_color)],
-                        'marker.cmax': [None, np.max(node_color)]
-                    }],
-                    label=column_name,
-                    method='restyle'
-                )
-            )
-        return column_color_buttons
-
-
-def _get_colorscale_buttons(colorscales):
-    colorscale_buttons = []
-    for colorscale in colorscales:
-        colorscale_buttons.append(
+        columns_to_color = range(data.shape[1])
+
+    node_color_map = list(map(lambda x: rgb2hex(get_cmap(colorscale)(x)),
+                              node_colors_color_variable))
+
+    column_color_buttons = [
+        dict(
+            args=[{
+                'marker.color': [None, node_color_map],
+                'marker.cmin': [None, np.min(node_colors_color_variable)],
+                'marker.cmax': [None, np.max(node_colors_color_variable)],
+                'hoverlabel.bgcolor': [None, node_color_map]
+            }],
+            label='color_variable',
+            method='restyle'
+        )
+    ]
+
+    for column in columns_to_color:
+        if is_data_dataframe:
+            column_values = data[column].to_numpy()
+        else:
+            column_values = data[:, column]
+        node_colors = get_node_summary(node_elements, column_values)
+        node_color_map = list(map(lambda x: rgb2hex(get_cmap(colorscale)(x)),
+                                  node_colors))
+
+        column_color_buttons.append(
             dict(
-                args=[{'marker.colorscale': [None, colorscale]}],
-                label=colorscale,
+                args=[{
+                    'marker.color': [None, node_color_map],
+                    'marker.cmin': [None, np.min(node_colors)],
+                    'marker.cmax': [None, np.max(node_colors)],
+                    'hoverlabel.bgcolor': [None, node_color_map]
+                }],
+                label='Column {}'.format(column),
                 method='restyle'
             )
         )
-    return colorscale_buttons
+    return column_color_buttons
+
+
+def _is_array_or_dataframe(color_variable, data):
+    """Determines whether color_variable is array or pandas dataframe."""
+    if hasattr(color_variable, 'dtype') or hasattr(color_variable, 'dtypes'):
+        if len(color_variable) != len(data):
+            raise ValueError(
+                "color_variable and data must have the same length.")
+        color_variable_kind = 'scalars'
+    elif hasattr(color_variable, 'transform'):
+        color_variable_kind = 'transformer'
+    elif hasattr(color_variable, 'fit_transform'):
+        color_variable_kind = 'fit_transformer'
+    elif callable(color_variable):
+        color_variable_kind = 'callable'
+    elif color_variable is None:
```

Comment:
Now that this function has been isolated in utils, I wonder if the correct string values for the last two cases should be 'none' and 'else' respectively.

Reply:
(Or something similarly generic.)

Reply:
Done as you suggest